akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Routers.scala

```diff
@@ -60,6 +59,13 @@ trait GroupRouter[T] extends Behavior[T] {
    */
   def withRoundRobinRouting(): GroupRouter[T]
 
+  /**
+   * Route messages by using consistent hashing.
+   *
+   * See [[akka.actor.typed.javadsl.GroupRouter#withConsistentHashingRouting]].
+   */
+  def withConsistentHashingRouting(virtualNodesFactor: Int, mapping: HashExtractor[T]): GroupRouter[T]
```

Comment:
do we gain much by defining the `HashExtractor` interface? compared with just a function `T => Array[Byte]`

Reply:
This was my suggestion to get a single SAM for both Java and Scala APIs

Reply:
but we already have javadsl and scaladsl so there wouldn't be any conflicts between function representations

The HashExtractor type could add some value in that it can explain more what it is than a function, but it also adds another thing which is just a glorified function (that could be explained right here instead)

Reply:
It started out as a `T => Any` and doing type check on the return when calling it to cover both `String` and `Array[Byte]`. I think that was clunky so recommended away from it suggesting a stricter type (or a sealed hierarchy).

What would your design suggestion be?

Reply:
I agree that the Any was vague and using serialization overkill. I'd suggest `T => String`.

Defining a sealed hierarchy could work but don't think it's worth complicating it.

Reply:
Will change to `T => String` then :)